### PR TITLE
add yq step action for scaffolder

### PR DESCRIPTION
**Pull request by punkle**
add yq step action for scaffolder

#### :heavy_check_mark: Checklist

- [ ] Added tests for new functionality and regression tests for bug fixes
- [x] Added changeset (run `yarn changeset` in the root)
- [ ] Screenshots of before and after attached (for UI changes)
- [x] Added or updated documentation (if applicable)

### CODE REVIEW

File: plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/yq/yq.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+      ctx.logger.info(`Attempting to yq with ${ctx.input.path}`);
+
+      const writePath =
+        ctx.input.writeOutputPath &&
+        resolveSafeChildPath(ctx.workspacePath, ctx.input.writeOutputPath);
+      if (!fs.existsSync(sourceFilepath)) {
+        throw new Error(`${messageSanitizer(sourceFilepath)} does not exist`);
+      }
+      try {
+        const initialResult = await execFilePromise('yq', [sourceFilepath]);
```
Comment (punkle) [inline plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/yq/yq.ts:109]: This type of code that depends on a cli command is very hard to test.

File: .changeset/neat-sheep-rest.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@roadiehq/scaffolder-backend-module-utils': patch
```
Comment (Xantier) [inline .changeset/neat-sheep-rest.md:2]: Is this a breaking change? It introduces an underlying command `yq` which is not one of the default ones in any sane distros

File: plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/yq/yq.ts (hunk 1 above)
Comment (Xantier) [reply plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/yq/yq.ts:109]: Does it have to be a cli command?

**Review by kissmikijr — CHANGES_REQUESTED** [review]
Can you elaborate a bit why do we need this command ? In general I don't like that we rely on a cli that needs to be installed on the host where this action runs. 
I think we should be able to do this with native js without other dependencies to be present on the system

File: plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/yq/yq.ts (hunk 1 above)
Comment (punkle) [reply plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/yq/yq.ts:109]: I havent found a library for this yet. Suggestions welcome.

File: .changeset/neat-sheep-rest.md (hunk 2 above)
Comment (punkle) [reply .changeset/neat-sheep-rest.md:2]: My thinking here is that you dont have to choose to use this action and current usage is not using it. So it is not breaking in that sense. Happy to make it breaking if you have a different perspective.

File: plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/yq/yq.ts (hunk 1 above)
Comment (Xantier) [reply plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/yq/yq.ts:109]: `yaml` on NPM would allow you to achieve this

**Comment by punkle** [comment]
The use case: I want to add an item to an array in a yaml file and then write it back to the file. I tried lodash.mergeWith but it does not work with nested items. This was suggested that it would be a richer solution in that it would allow lots of other types of interaction with yaml data.

File: plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/yq/yq.ts (hunk 1 above)
Comment (punkle) [reply plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/yq/yq.ts:109]: I think its the merging nested arrays part that Ive trouble with. How would that be surfaced via an action?

e.g.

if I have a yaml file containing:

```yaml
person:
   numbers:
    - 555123456
    - 555123457
```

How do I form an action to append a new number to that list and write it back to the file?

With the action I have just created I can do that with:

```yaml
    - id: yamlmerge
      name: YAML Merge
      action: roadiehq:utils:yq
      input:
        path: person.yaml
        writeOutputPath: person.yaml
        query:
          - '(.person.numbers = .person.numbers + 555123458)'
```

Comment (punkle) [reply plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/yq/yq.ts:109]: It also allows for deleting that phone number later.

**Pull request closed without merging**